### DISCUSSION TITLE
Use included prism to highlight custom code block

### DISCUSSION THREAD

**Discussion by TheZoker** [facebook/docusaurus#4379] [Q&A]
Hi there,

I'm currently locking for a way to use the included prism to highlight code withing a custom div element. 

Something like
```html
<div class="code">...</div
```
and then
```js
var code = document.getElementsByClassName('code');
Array.prototype.map.call(code, function(elem) {
   elem.highlight();
});
```

I know I can also include prism as a dependency, but I would rather just use the included syntax highlighting. 
Does the prism script just look for a certain class or is there any method I can use to highlight custom elements?

Thanks!

**Comment by armano2** [comment]
you could use `@theme/CodeBlock` instead 
https://github.com/facebook/docusaurus/blob/f51f21965cd300c5467b0214404bf718254f17d1/packages/docusaurus-theme-classic/src/types.d.ts#L53-L62

```tsx
import CodeBlock from '@theme/CodeBlock'

<CodeBlock metastring="js">...</CodeBlock>
```

personally i don't like how api of this component is written as it should have explicit properties for `language`, `title`, `highlight` instead of `metastring`

**Reply by slorber** [reply]
agree, we should rather have 1 comp wired to MDX and one component with a proper API surface to be more easily imported/used in MDX

**Reply by TheZoker** [reply]
Thanks, works! But my content is changing dynamically. Is there any option to apply the code highlighting on data change or is this currently only possible with the content that is provided on load?
